plugins/ltac2/tac2core.ml

```diff
@@ -394,10 +401,24 @@ let () = binop "int_compare" Int.compare
 let () = binop "int_add" (+)
 let () = binop "int_sub" (-)
 let () = binop "int_mul" ( * )
-
-let () = define1 "int_neg" int begin fun m ->
-  return (Value.of_int (~- m))
-end
+let () = define2 "int_div" int int begin fun m n ->
+  if n == 0 then throw err_division_by_zero
```

Comment:
I don't think it matters much for a `n==0`. I would think that entering a try catch block is also not for free and is likely comparable to an `if n==0`, but I must admit I never looked into the code OCaml generates for a try catch. I think this matters more if the if condition is not entirely trivial, say a lookup of a key in a tree based dictionary.

If a try catch is entirely for free on OCaml, I can change it.

Reply:
It doesn't matter. That said, I think you should rewrite the `wrap` because it's useless and introduces a small cost there. This function should only be used when the thunk can produce side-effects.

Reply:
Yes right - since I check for 0 before, the division cannot result in an exception any more. I will change it.

Reply:
I pushed the changes.